src/auth/feedback-secret.guard.ts

```diff
@@ -0,0 +1,24 @@
+import {
+  Injectable,
+  CanActivate,
+  ExecutionContext,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { Env } from 'src/env';
+
+@Injectable()
+export class FeedbackSecretGuard implements CanActivate {
+  constructor() {}
+
+  canActivate(context: ExecutionContext): boolean {
+    const request = context.switchToHttp().getRequest();
+    const secret = Env.SECRET.FEEDBACK;
+    const requestSecret = request.headers['x-feedback-secret'];
```

Comment:
PullProofAI

There is a potential security vulnerability in the way the feedback secret is being compared. Directly comparing secrets in this manner can expose the application to timing attacks.

#### Recommendation
Use a secure comparison method like a constant-time comparison function to compare the secrets to mitigate timing attacks.

```diff
-     if (requestSecret && requestSecret === secret) {
+     if (requestSecret && secureCompare(requestSecret, secret)) {
```

### Critical Issue



<!-- This is an auto-generated comment by PullProofAI -->